packages/@react-spectrum/button/test/ButtonGroup.test.js

```diff
@@ -97,7 +97,288 @@ describe('ButtonGroup', function () {
     expect(group).toHaveAttribute('aria-disabled', 'true');
   });
 
-  it('SideNav handles single selection', function () {
+  it.each`
+    Name               | ComponentGroup   | Component       | props
+    ${'ButtonGroup'}   | ${ButtonGroup}   | ${ActionButton} | ${{}}
+  `('$Name shifts button focus on left/right keyboard arrow click (ltr + horizontal)', function ({ComponentGroup, Component, props}) {
+    let tree = render(
+      <Provider theme={theme} locale="de-DE">
+        <ComponentGroup {...props} >
+          <Component data-testid="button-1">Click me 1</Component>
+          <Component data-testid="button-2">Click me 2</Component>
+          <Component data-testid="button-3">Click me 3</Component>
+        </ComponentGroup>
+      </Provider>
+    );
+
+    let button1 = tree.getByTestId('button-1');
+    let button2 = tree.getByTestId('button-2');
+    let button3 = tree.getByTestId('button-3');
+
+    triggerPress(button1);
+    expect(button1).toHaveAttribute('tabIndex', '0');
+
+    fireEvent.keyDown(button1, {key: 'ArrowRight'});
+    expect(button1).toHaveAttribute('tabIndex', '-1');
+    expect(button2).toHaveAttribute('tabIndex', '0');
+
+    fireEvent.keyDown(button2, {key: 'ArrowLeft'});
+    expect(button1).toHaveAttribute('tabIndex', '0');
+    expect(button2).toHaveAttribute('tabIndex', '-1');
+
+    fireEvent.keyDown(button1, {key: 'ArrowLeft'});
+    expect(button3).toHaveAttribute('tabIndex', '0');
+    expect(button1).toHaveAttribute('tabIndex', '-1');
+  });
+
+  it.each`
+    Name               | ComponentGroup   | Component       | props
+    ${'ButtonGroup'}   | ${ButtonGroup}   | ${ActionButton} | ${{}}
```

Comment:
```
it.each`
    Name               | ComponentGroup   | Component       | props   | orders
    ${'ButtonGroup left/right ltr+horizontal'}   | ${ButtonGroup}   | ${ActionButton} | ${{locale: 'de-DE'}} | ${[{action: triggerPress, result: [0, -1, -1]}, {action: fireEvent.keyDown, result: [-1, 0, -1]}, {action: fireEvent.keydown, result: [0, -1, -1]}, {action: fireEvent.keyDown, result: [-1, -1, 0]}]}
    // for brevity i'm not writing out full orders
    ${'ButtonGroup up/down ltr+horizontal'}   | ${ButtonGroup}   | ${ActionButton} | ${{locale: 'de-DE'}} | ${[[0, -1, -1], [-1, 0, -1], [0, -1, -1], [-1, -1, 0]]}
    ${'ButtonGroup left/right rtl+horizontal'}   | ${ButtonGroup}   | ${ActionButton} | ${{locale: 'ar-AE'}} | ${[[0, -1, -1], [-1, -1, 0], [0, -1, -1], [-1, 0, -1]]}
```	   
etc

and then inside
```
// this function top of module
function verifyResult(buttons, values) {
  buttons.forEach((button, i) => {
    expect(button).toHaveAttribute('tabIndex', values[i]);
  })
}

let buttons = [tree.getByTestId('button-1'), tree.getByTestId('button-2'), tree.getByTestId('button-3')];
expect(buttons[0]).toHaveAttribute('tabIndex', '0');

orders.forEach(({action, result}) => {
  action(target); // how you get target up to you, maybe you can just always do it to the button with tabIndex 0?
  verifyResult(buttons, result);
});
```

Reply:
Doing something like this should make it easier to see the differences between the modes